src/main/java/picard/sam/markduplicates/UmiAwareDuplicateSetIterator.java

```diff
@@ -156,17 +153,19 @@ private void process(final DuplicateSet set) {
                             throw new PicardException("UMIs of differing lengths were found.");
                         }
                     }
-//                    metrics.updateUmiRecordMetrics(currentUmi, inferredUmi);
+
+                    // Update UMI metrics associated with each record
                     metrics.OBSERVED_BASE_ERRORS += hammingDistance(currentUmi, inferredUmi);
                     observedUmiBases += currentUmi.length();
-                    observedUmis.increment(currentUmi);
-                    inferredUmis.increment(inferredUmi);
+                    metrics.observedUmis.increment(currentUmi);
```

Comment:
Seeing how this is used, I would reccomend changing `observedUmis` and `inferredUmis` to private members of the metric and adding a function .addUmiObservation that takes both currentUmi and inferredUmi as input. this function should also update a private long `UmiBasesObserved` member to use in `calculateDerivedMetrics()`